storeshare_api/app/models/conversation.rb

```diff
@@ -0,0 +1,9 @@
+class Conversation < ActiveRecord::Base
+  belongs_to :renter        #, foreign_key: "sender_id"
+  belongs_to :subletter        #, foreign_key:"recipient_id"
+  has_many :messages, dependent: :destroy
+  validates_uniqueness_of :renter_id, :scope => :subletter_id  #therefore there can be only one Conversation between a specific renter and subletter
```

Comment:
if a subletter has multiple listings posted on the website, do we still want there to be only one conversation for a renter-subletter pair across all listings, or should we instead validate the uniqueness of (renter, subletter, listing) triples

Reply:
I feel that it would be unnecessary complication but I would like to hear Alex's opinion about this.

Reply:
I think either could be fine, depending on intent! The way airbnb does it: each reservation has a corresponding messaging thread, specific to that reservation (even if a customer has multiple reservations with the same host). 

In terms of implementation complexity-- I don't think using a triple `(renter, subletter, listing / inquiry / rental contract / etc)` versus `(renter, subletter)` at a database level would be too bad

If there is time, I think we should eventually do per-lease messaging, but for iteration 2 / tonight's deadline just using the tuple instead of the triple is fine.